consensus/spos/bls/subroundEndRound.go

```diff
@@ -553,48 +560,100 @@ func (sr *subroundEndRound) aggregateSigsAndHandleInvalidSigners(bitmap []byte)
 	return bitmap, sig, nil
 }
 
-func (sr *subroundEndRound) verifyNodesOnAggSigFail() ([]string, error) {
+// Method to let thread continue its execution so that another goroutine can be called
+func (sr *subroundEndRound) checkGoRoutinesThrottler(ctx context.Context) error {
+	for {
+
+		if sr.signatureThrottler.CanProcess() {
+			break
+		}
+
+		select {
+		case <-time.After(time.Millisecond):
+			continue
+		case <-ctx.Done():
+			return spos.ErrTimeIsOut
+		}
+	}
+	return nil
+}
+
+// Method that implements parallel signature verification
+func (sr *subroundEndRound) signatureVerification(i int, pk string, sigShare []byte) error {
+
+	isSuccessful := true
```

Comment:
no need for this boolean.. in case of err, return err and add another log.trace, in case of success, simply return nil